clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/model/GoogleApplication2.groovy

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.google.model
+
+import com.fasterxml.jackson.annotation.JsonPropertyOrder
+import com.netflix.spinnaker.clouddriver.model.Application
+import groovy.transform.Canonical
+import groovy.transform.CompileStatic
+import groovy.transform.EqualsAndHashCode
+
+@CompileStatic
+@EqualsAndHashCode(includes = ["name"])
+class GoogleApplication2 {
+  String name
+
+  View getView() {
+    new View()
+  }
+
+  @JsonPropertyOrder(alphabetic = true)
+  @Canonical
+  class View implements Application {
+    String name = GoogleApplication2.this.name
+    Map<String, String> attributes = Collections.synchronizedMap([:])
```

Comment:
Do you need a `synchronizedMap` here? Are multiple threads potentially accessing/mutating the `attributes`?